File: dqops/src/main/java/com/dqops/core/domains/LocalDataDomainModel.java
/*
 * Copyright © 2021-Present DQOps, Documati sp. z o.o. (support@dqops.com)
 *
 * This file is licensed under the Business Source License 1.1,
 * which can be found in the root directory of this repository.
 *
 * Change Date: This file will be licensed under the Apache License, Version 2.0,
 * four (4) years from its last modification date.
 */
package com.dqops.core.domains;

import com.dqops.metadata.settings.domains.LocalDataDomainSpec;
import com.dqops.utils.docs.generators.SampleValueFactory;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonPropertyDescription;
import com.fasterxml.jackson.databind.PropertyNamingStrategies;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import io.swagger.annotations.ApiModel;
import lombok.Data;

import java.time.*;


/**
 * Model that describes a local data domain. It is also used to create new data domains.
 */
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonNaming(PropertyNamingStrategies.SnakeCaseStrategy.class)
@ApiModel(value = "LocalDataDomainModel", description = "Model that describes a local data domain. It is also used to create new data domains.")
public class LocalDataDomainModel {
    /**
     * Data domain name.
     */
    @JsonPropertyDescription("Data domain name.")
    private String domainName;

    /**
     * Data domain display name.
     */
    @JsonPropertyDescription("Data domain display name (user-friendly name).")
    private String displayName;

    /**
     * Data domain creation time.
     */
    @JsonPropertyDescription("Data domain creation time.")
    private Instant createdAt;

    /**
     * Enables the job scheduler for this domain.
     */
    @JsonPropertyDescription("Enables the job scheduler for this domain.")
    private boolean enableScheduler = true;

    public LocalDataDomainModel() {
    }

    /**
     * Creates a data domain model from a specification object.
     * @param localDataDomainSpec Local data domain specification as a source.
     * @return Data domain model.
     */
    public static LocalDataDomainModel createFromSpec(LocalDataDomainSpec localDataDomainSpec) {
        LocalDataDomainModel localDataDomainModel = new LocalDataDomainModel() {{
            setDomainName(localDataDomainSpec.getDataDomainName());
            setDisplayName(localDataDomainSpec.getDisplayName());
            setCreatedAt(localDataDomainSpec.getCreatedAt());
            setEnableScheduler(localDataDomainSpec.isEnableScheduler());
        }};

        return localDataDomainModel;
    }

    /**
     * Creates a local data domain specification from the model.
     * @return Local data domain specification object.
     */
    public LocalDataDomainSpec toLocalDataDomainSpec() {
        LocalDataDomainSpec localDataDomainSpec = new LocalDataDomainSpec() {{
            setDataDomainName(this.getDataDomainName());
            setDisplayName(this.getDisplayName());
            setCreatedAt(this.getCreatedAt() != null ? this.getCreatedAt() : Instant.now()); // we are using the current time otherwise
            setEnableScheduler(this.isEnableScheduler());
        }};

        return localDataDomainSpec;
    }

    /**
     * Documentation samples factory.
     */
    public static class LocalDataDomainModelSampleFactory implements SampleValueFactory<LocalDataDomainModel> {
        @Override
        public LocalDataDomainModel createSample() {
            return new LocalDataDomainModel() {{
                setDomainName("sales");
                setDisplayName("Sales data domain");
                setCreatedAt(OffsetDateTime.of(LocalDateTime.of(2024, Month.SEPTEMBER, 14, 18, 33), ZoneOffset.UTC).toInstant());
            }};
        }
    }
}

File: distribution/python/dqops/client/models/local_data_domain_model.py
from typing import Any, Dict, List, Type, TypeVar, Union

from attrs import define as _attrs_define
from attrs import field as _attrs_field

from ..types import UNSET, Unset

T = TypeVar("T", bound="LocalDataDomainModel")


@_attrs_define
class LocalDataDomainModel:
    """Model that describes a local data domain. It is also used to create new data domains.

    Attributes:
        domain_name (Union[Unset, str]): Data domain name.
        display_name (Union[Unset, str]): Data domain display name (user-friendly name).
        created_at (Union[Unset, int]): Data domain creation time.
        enable_scheduler (Union[Unset, bool]): Enables the job scheduler for this domain.
    """

    domain_name: Union[Unset, str] = UNSET
    display_name: Union[Unset, str] = UNSET
    created_at: Union[Unset, int] = UNSET
    enable_scheduler: Union[Unset, bool] = UNSET
    additional_properties: Dict[str, Any] = _attrs_field(init=False, factory=dict)

    def to_dict(self) -> Dict[str, Any]:
        domain_name = self.domain_name
        display_name = self.display_name
        created_at = self.created_at
        enable_scheduler = self.enable_scheduler

        field_dict: Dict[str, Any] = {}
        field_dict.update(self.additional_properties)
        field_dict.update({})
        if domain_name is not UNSET:
            field_dict["domain_name"] = domain_name
        if display_name is not UNSET:
            field_dict["display_name"] = display_name
        if created_at is not UNSET:
            field_dict["created_at"] = created_at
        if enable_scheduler is not UNSET:
            field_dict["enable_scheduler"] = enable_scheduler

        return field_dict

    @classmethod
    def from_dict(cls: Type[T], src_dict: Dict[str, Any]) -> T:
        d = src_dict.copy()
        domain_name = d.pop("domain_name", UNSET)

        display_name = d.pop("display_name", UNSET)

        created_at = d.pop("created_at", UNSET)

        enable_scheduler = d.pop("enable_scheduler", UNSET)

        local_data_domain_model = cls(
            domain_name=domain_name,
            display_name=display_name,
            created_at=created_at,
            enable_scheduler=enable_scheduler,
        )

        local_data_domain_model.additional_properties = d
        return local_data_domain_model

    @property
    def additional_keys(self) -> List[str]:
        return list(self.additional_properties.keys())

    def __getitem__(self, key: str) -> Any:
        return self.additional_properties[key]

    def __setitem__(self, key: str, value: Any) -> None:
        self.additional_properties[key] = value

    def __delitem__(self, key: str) -> None:
        del self.additional_properties[key]

    def __contains__(self, key: str) -> bool:
        return key in self.additional_properties
